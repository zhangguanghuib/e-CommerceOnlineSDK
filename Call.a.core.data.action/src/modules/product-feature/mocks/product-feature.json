{
  "id": "R1Module1",
  "config": {
    "imageAlignment": "left",
    "productTitle": "Black Striped Tie",
    "productDetails": "Narrow stripes stretch across a tie made from Italian silk.",
    "productImage": {
      "src": "https://images-au-prod.cms.commerce.dynamics.com/cms/api/rrjxclpqcv/imageFileData/search?fileName=/Products%2F91091_000_001.png&amp;fallback=/Products/91091_000_001.jpg,/Products/91091_000_001.png,/Products/91091_000_001.webp&amp;m=6&amp;q=80&amp;cropfocalregion=true" ,
      "altText": "Black Striped Tie"
    },
    "buttonText": "Buy Now",
    "productIds": "68719495953"
  },
  "typeName": "product-feature"
}
