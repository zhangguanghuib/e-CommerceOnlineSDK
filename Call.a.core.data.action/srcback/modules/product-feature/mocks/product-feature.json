{
  "id": "R1Module1",
  "config": {
    "imageAlignment": "left",
    "productTitle": "Black Striped Tie",
    "productDescription": "Narrow stripes stretch across a tie made from Italian silk.",
    "productImage": {
      "src": "https://images-au-prod.cms.commerce.dynamics.com/cms/api/rrjxclpqcv/imageFileData/search?fileName=/Products%2F91091_000_001.png&fallback=/Products/91091_000_001.jpg,/Products/91091_000_001.png,/Products/91091_000_001.webp&m=6&q=80&cropfocalregion=true",
      "alt": "Black Striped Tie"
    },
    "buttonText": "Buy Now",
    "productIds": "68719495953"
  },
  "typeName": "product-feature"
}