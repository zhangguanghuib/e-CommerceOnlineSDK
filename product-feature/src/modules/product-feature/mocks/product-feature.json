{
  "id": "R1Module1",
  "config": {
    "imageAlignment": "left",
    "productTitle": "Retro Horn Rimmed Keyhole Nose Bridge Round Sunglasses",
    "productDescription": "High-quality and pioneered with the perfect blend of timeless classic and modern technology with hint of old school glamor.",
    "productImage": {
      "src": "https://images-au-prod.cms.commerce.dynamics.com/cms/api/rrjxclpqcv/imageFileData/ML2jfH?pubver=1",
      "alt": "Retro Horn Rimmed Keyhole Nose Bridge Round Sunglasses"
    },
    "buttonText": "Shop Now"
  },
  "typeName": "product-feature"
}