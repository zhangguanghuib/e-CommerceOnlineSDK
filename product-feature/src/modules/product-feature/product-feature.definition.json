{
    "$type": "contentModule",
    "friendlyName": "product-feature",
    "name": "product-feature",
    "description": "Feature module used to highlight a product.",
    "categories": [
        "storytelling"
    ],
    "tags": [
        ""
    ],
    "dataActions": {        
    },    
    "config": {
        "imageAlignment":{
            "friendlyName": "Image Alignment",
            "description": "Set the desired alignment for the image, either left or right.",
            "type": "string",
            "enum":{ "left": "Left", "right": "Right" },
            "default": "left",
            "scope": "module",
            "group": "layout Properties"
        },
        "productTitle":{
            "type": "string",
            "friendlyName": "Product Title",
            "description": "Product placement title."
        }, 
        "productDetails":{
            "type": "richText",
            "friendlyName": "Product Details",
            "description": "Rich text representing the featured product details."
        },
        "productImage":{
            "type": "image",
            "friendlyName": "Product Image",
            "description": "Image representing the featured product."
        },
        "buttonText":{
            "type": "string",
            "friendlyName": "Button Text",
            "description": "Text to show on the call to action button."
        }
      }
}