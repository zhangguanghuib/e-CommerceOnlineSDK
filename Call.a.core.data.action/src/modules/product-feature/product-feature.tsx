/*!
 * Copyright (c) Microsoft Corporation.
 * All rights reserved. See LICENSE in the project root for license information.
 */

import * as React from 'react';

import { IProductFeatureData } from './product-feature.data';
import { imageAlignment, IProductFeatureProps } from './product-feature.props.autogenerated';

export interface IProductFeatureViewProps extends IProductFeatureProps<IProductFeatureData> {
    productName: string;
    productInfo: string;
    productImageUrl: string;
    productPrice: string;
    buttonInfo: string;
    alignment: imageAlignment;
}

/**
 *
 * ProductFeature component
 * @extends {React.PureComponent<IProductFeatureProps<IProductFeatureData>>}
 */
class ProductFeature extends React.PureComponent<IProductFeatureProps<IProductFeatureData>> {
    public render(): JSX.Element | null {
        const {
            config,
            data: { products }
        } = this.props;

        // set default product info values
        let ProductName = config.productTitle ? config.productTitle : 'No product name defined';
        let ProductInfo = config.productDetails ? config.productDetails.toString() : 'No product details defined';
        let ProductImageUrl = config.productImage ? config.productImage.src : '';
        const ButtonInfo = config.buttonText ? config.buttonText : 'No button text defined';
        let ProductPrice = '$129';

        // if product details were returned from Retail server, then we'll use that info instead of config values
        if (products && products.length) {
            ProductName = products[0].result && products[0].result.Name ? products[0].result.Name : 'Invalid product data';
            ProductInfo = products[0].result && products[0].result.Description ? products[0].result.Description : 'Invalid product data';
            ProductImageUrl = products[0].result && products[0].result.PrimaryImageUrl ? products[0].result.PrimaryImageUrl : '';
            ProductPrice = products[0].result && products[0].result.Price ? `$ ${products[0].result.Price}` : '';
        }

        const ProductFeatureViewProps = {
            ...this.props,
            productName: ProductName,
            productInfo: ProductInfo,
            productImageUrl: ProductImageUrl,
            productPrice: ProductPrice,
            buttonInfo: ButtonInfo,
            alignment: config.imageAlignment
        };

        return this.props.renderView(ProductFeatureViewProps);
    }
}

export default ProductFeature;
