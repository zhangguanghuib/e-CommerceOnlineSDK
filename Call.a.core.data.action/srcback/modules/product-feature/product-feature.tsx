/*!
 * Copyright (c) Microsoft Corporation.
 * All rights reserved. See LICENSE in the project root for license information.
 */

import * as React from 'react';

import { IProductFeatureData } from './product-feature.data';
import { imageAlignment, IProductFeatureProps } from './product-feature.props.autogenerated';

export interface IProductFeatureViewProps extends IProductFeatureProps<IProductFeatureData> {
    productName: string;
    productInfo: string;
    productImageUrl: string;
    productPrice: string;
    buttonInfo: string;
    alignment: imageAlignment;
}

/**
 *
 * ProductFeature component
 * @extends {React.PureComponent<IProductFeatureProps<IProductFeatureData>>}
 */
class ProductFeature extends React.PureComponent<IProductFeatureProps<IProductFeatureData>> {
    public async render(): Promise<JSX.Element | null> {
        const {
            config,
            data: { products }
        } = this.props;

        // set default product info values
        let ProductName = config.productTitle ? config.productTitle : 'No product name defined';
        let ProductInfo = config.productDetails ? config.productDetails.toString() : 'No product details defined';
        let ProductImageUrl = config.productImage ? config.productImage.src : '';
        const ButtonInfo = config.buttonText ? config.buttonText : 'No button text defined';
        let ProductPrice = '$129.99';

        if (products && products.length > 0) {
            ProductName = products[0].result && products[0].result.Name ? products[0].result.Name : ProductName;
            ProductInfo = products[0].result && products[0].result.Description ? products[0].result.Description : ProductInfo;
            ProductImageUrl =
                products[0].result && products[0].result.PrimaryImageUrl ? products[0].result.PrimaryImageUrl : ProductImageUrl;
            ProductPrice = products[0].result && products[0].result.Price ? `${products[0].result.Price}` : ProductPrice;
        }

        const ProductFeatureViewProps = {
            ...this.props,
            productName: ProductName,
            productInfo: ProductInfo,
            productImageUrl: ProductImageUrl,
            productPrice: ProductPrice,
            buttonInfo: ButtonInfo,
            alignment: config.imageAlignment
        };
        return this.props.renderView(ProductFeatureViewProps);
    }
}

export default ProductFeature;
