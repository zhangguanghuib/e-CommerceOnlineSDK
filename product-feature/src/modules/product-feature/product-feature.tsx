/*!
 * Copyright (c) Microsoft Corporation.
 * All rights reserved. See LICENSE in the project root for license information.
 */

import * as React from 'react';

import { IProductFeatureData } from './product-feature.data';
import { imageAlignment, IProductFeatureProps } from './product-feature.props.autogenerated';

export interface IProductFeatureViewProps extends IProductFeatureProps<IProductFeatureData> {
    productName: string;
    productInfo: string;
    productImageUrl: string;
    productPrice: string;
    buttonInfo: string;
    alignment: imageAlignment;
}

/**
 *
 * ProductFeature component
 * @extends {React.PureComponent<IProductFeatureProps<IProductFeatureData>>}
 */
class ProductFeature extends React.PureComponent<IProductFeatureProps<IProductFeatureData>> {
    public render(): JSX.Element | null {
        const { config } = this.props;
        const ProductName = config.productTitle ? config.productTitle : 'No product name defined';
        const ProductInfo = config.productDetails ? config.productDetails.toString() : 'No product details defined';
        const ProductImageUrl = config.productImage ? config.productImage.src : '';
        const ButtonInfo = config.buttonText ? config.buttonText : 'No button text defined';
        const ProductPrice = '$129.99';

        const ProductFeatureViewProps = {
            ...this.props,
            productName: ProductName,
            productInfo: ProductInfo,
            productImageUrl: ProductImageUrl,
            productPrice: ProductPrice,
            buttonInfo: ButtonInfo,
            alignment: config.imageAlignment
        };
        return this.props.renderView(ProductFeatureViewProps);
    }
}

export default ProductFeature;
