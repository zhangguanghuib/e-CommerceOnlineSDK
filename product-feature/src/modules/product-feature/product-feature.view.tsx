/*!
 * Copyright (c) Microsoft Corporation.
 * All rights reserved. See LICENSE in the project root for license information.
 */

import * as React from 'react';
import { IProductFeatureViewProps } from './product-feature';
import { imageAlignment } from './product-feature.props.autogenerated';

const _renderImage = (productImageUrl: string, productname: string): JSX.Element => {
    return <img src={productImageUrl} alt={productname} className='img-fluid p-3' />;
};

const _renderInfo = (
    productName: string,
    productInfo: string,
    productPrice: string,
    buttonInfo: string,
    productTitledId: string
): JSX.Element => {
    return (
        <div className='container'>
            <h2 id={productTitledId}>{productName}</h2>
            <p>{productInfo}</p>
            <p>{productPrice}</p>
            <button type='button' className='btn btn-primary'>
                {buttonInfo}
            </button>
        </div>
    );
};

export default (props: IProductFeatureViewProps) => {
    const { productName, productInfo, productImageUrl, productPrice, buttonInfo, alignment } = props;

    let left;
    let right;
    if (alignment === imageAlignment.left) {
        left = _renderImage(productImageUrl, productName);
        right = _renderInfo(productName, productInfo, productPrice, buttonInfo, 'product_Title_1');
    } else {
        left = _renderInfo(productName, productInfo, productPrice, buttonInfo, 'product_Title_2');
        right = _renderImage(productImageUrl, productName);
    }

    return (
        <div className='row align-items-center'>
            <div className='col-md-6'>{left}</div>
            <div className='col-md-6'>{right}</div>
        </div>
    );
};
