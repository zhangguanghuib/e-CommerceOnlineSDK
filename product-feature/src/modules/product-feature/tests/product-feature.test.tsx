/*!
 * Copyright (c) Microsoft Corporation.
 * All rights reserved. See LICENSE in the project root for license information.
 */

import { buildMockModuleProps } from '@msdyn365-commerce/core-internal';
/// <reference types="jest" />

import * as React from 'react';
import * as renderer from 'react-test-renderer';

import ProductFeature from '../product-feature';
import { IProductFeatureData } from '../product-feature.data';
import { imageAlignment, IProductFeatureConfig, IProductFeatureProps } from '../product-feature.props.autogenerated';

const mockData: IProductFeatureData = {
    actionResponse: {
        text: 'Sample Response Data'
    }
};

const mockConfig: IProductFeatureConfig = {
    imageAlignment: imageAlignment.left,
    productTitle: 'Retro Horn Rimmed Keyhole Nose Bridge Round Sunglasses',
    productDetails:
        'High-quality and pioneered with the perfect blend of timeless classic and modern technology with hint of old school glamor.',
    productImage: {
        src: 'https://bit.ly/33cMGxr',
        altText: 'Retro Horn Rimmed Keyhole Nose Bridge Round Sunglasses'
    },
    buttonText: 'Buy Now'
};

const mockActions = {};

describe('ProductFeature', () => {
    let moduleProps: IProductFeatureProps<IProductFeatureData>;
    beforeAll(() => {
        moduleProps = buildMockModuleProps(mockData, mockActions, mockConfig) as IProductFeatureProps<IProductFeatureData>;
    });
    it('renders correctly', () => {
        const component: renderer.ReactTestRenderer = renderer.create(<ProductFeature {...moduleProps} />);
        const tree: renderer.ReactTestRendererJSON | renderer.ReactTestRendererJSON[] | null = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
