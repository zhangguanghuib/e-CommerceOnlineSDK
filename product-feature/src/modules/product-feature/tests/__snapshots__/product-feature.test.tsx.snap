// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductFeature renders correctly 1`] = `null`;
